test/cmake/modules/SwiftTestUtils.cmake

```diff
@@ -29,6 +29,8 @@ function(get_swift_test_versioned_target_triple variant_triple_out_var sdk arch
   if(build_flavor STREQUAL "ios-like")
     # Use the macCatalyst target triple and compiler resources for the iOS-like build flavor.
     set(variant_triple "${arch}-apple-ios13.0-macabi")
+  elseif(${sdk} STREQUAL ANDROID)
+    get_versioned_target_triple(variant_triple ${sdk} ${arch} ${SWIFT_ANDROID_API_LEVEL})
```

Comment:
I think this fixes your remaining issues with finding the Android C runtime and other libraries.